### PR TITLE
feat: query RegisteredTokenMarkets

### PR DESCRIPTION
**Pull request by toteki**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new service to query registered tokens and their market summaries.
  - Users can now view token market information within the app.

- **Documentation**
  - Added new message types for better querying and response structure in token market services.

- **Tests**
  - Implemented tests for the new token market querying functionality.

### CODE REVIEW

**Review by gsk967 — APPROVED** [review]
LGTM

**Comment by gsk967** [comment]
@toteki  WDYT  may be we can add this query as optional integration (customizable query) to binary

**Comment by toteki** [comment]
Since this doesn't iterate over users, it doesn't need to be enabled by flag like the `-l` queries. Should be safe as-is

File: proto/umee/leverage/v1/query.proto [hunk 1]
```diff
@@ -24,6 +24,12 @@ service Query {
     option (google.api.http).get = "/umee/leverage/v1/registered_tokens";
   }
 
+  // RegisteredTokenMarkets queries for all the registered tokens and their market summaries.
+  rpc RegisteredTokenMarkets(QueryRegisteredTokenMarkets)
+      returns (QueryRegisteredTokenMarketsResponse) {
+    option (google.api.http).get = "/umee/leverage/v1/registered_token_markets";
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/query.proto:30]:
```suggestion
  // RegisteredTokensWithMarkets queries for all the registered tokens and their market summaries.
  rpc RegisteredTokensWithMarkets(QueryRegisteredTokensWithMarkets)
      returns (QueryRegisteredTokensWithMarketsResponse) {
    option (google.api.http).get = "/umee/leverage/v1/registered_tokens_with_markets";
```

File: x/leverage/keeper/grpc_query.go [hunk 2]
```diff
@@ -62,6 +62,35 @@ func (q Querier) RegisteredTokens(
...
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	tokens := q.Keeper.GetAllRegisteredTokens(ctx)
+	markets := []types.TokenMarket{}
+
+	for _, token := range tokens {
+		marketSumnmary, err := q.MarketSummary(goCtx, &types.QueryMarketSummary{Denom: token.BaseDenom})
+		if err != nil {
+			// absorb overall query error into struct, which may be empty, but proceed with this query
+			marketSumnmary.Errors += err.Error()
```
Comment (robert-zaremba) [inline x/leverage/keeper/grpc_query.go:81]: why do we return errors as a String as a part of the "normal" object response?

**Review by robert-zaremba — APPROVED** [review]
utACK. Left one comment to discuss the API

File: x/leverage/keeper/grpc_query.go (hunk 2 above)
Comment (toteki) [reply x/leverage/keeper/grpc_query.go:81]: Mimics `MarketSummary` behavior.

This was before `errors.Join` existed, and it runs multiple calculations where we need to return the fields we can even if some of the others error. The string field kept the API simple (always a known, readable type for debugging)

**Pull request merged**